### DISCUSSION TITLE
Getting class/method names in a ResponseExceptionMapper

### DISCUSSION THREAD

**Discussion by martin-kofoed-jyskebank-dk** [quarkusio/quarkus#33322] [Q&A] [answered]
Hi,

Is it possible to somehow get the class and method name of a REST reactive client from within the `toThrowable()` method of a `ResponseExceptionMapper`? All the injectable context objects seem to only carry the API information which is not sufficient in a situation where one out of multiple client calls returns >400. I would like to add information to my error handling about the exact client endpoint called. Perhaps this could be achieved by adding a `toThrowable(Response response, Method method)` signature to the `ResponseExceptionMapper` interface?

Of cource I could handle response mapping in the client itself, but I was looking for a generic solution.

**Comment by Postremus** [comment]
Can you try if the `jakarta.ws.rs.container.ResourceInfo` works for your use case?
It has getResourceMethod and getResourceClass.

**Reply by martin-kofoed-jyskebank-dk** [reply]
Yes, that is one of the things I already tried. It works fine, but it only returns the API class/method. Not the reactive rest client. 

I have this call structure:

```
                            /--[RestClient1]-->
--[APIClass]--[ServiceClass]--[RestClient2]-->
                            \--[RestClient3]-->
```

So, when a response in ANY of the three RestClients fail (>400), and my provided, generic `ResponseExceptionMapper` kicks in, I'm only able to see which method/path in `APIClass` was called. I don't know if this works as designed, but I think it would make sense to be able to see exactly which method in which RestClient returned the error response.

**Reply by martin-kofoed-jyskebank-dk** [reply]
If I - on the other hand - use this approach in each of the RestClients:

```java
@ClientExceptionMapper
static RuntimeException toException(Response response, Method method) {
    ...
}
```
.. then I'm able to see exactly which method fails. But it's not a generic solution, and I have to paste a lot of code around.

**Reply by luca-bassoricci** [reply, marked as answer]
I was looking around client filer and found `io.quarkus.rest.client.reactive.runtime.ResteasyReactiveResponseExceptionMapper` which owns the method `T toThrowable(Response response, RestClientRequestContext context);`. 
May you register a client-filter to inject all necessary informations into request-context and with a custom exception-mapper of reactive type extract necessary information for error handling

**Reply by martin-kofoed-jyskebank-dk** [reply]
@luca-bassoricci - Actually, implementing `ResteasyReactiveResponseExceptionMapper` instead of `ResponseExceptionMapper` worked as a charm! Then it was just a matter of getting client info:

```java
String clientInfo = context.getInvokedMethod().getDeclaringClass().getSimpleName() + "::" + context.getInvokedMethod().getName();
```

.. and add it to the exception returned.

I think this option might be worth mentioning in the [Reactive REST Client documentation](https://quarkus.io/guides/rest-client-reactive#exception-handling)?

Thanks for pointing in the right direction.

**Reply by luca-bassoricci** [reply]
Probably @geoand can give us a more detailed response about the reasons behind not exposing in documentation this interfaces.

**Reply by geoand** [reply]
IIRC, it has already been documented in `main` by @Sgitario
